lib/grape/api/instance.rb

```diff
@@ -198,19 +198,21 @@ def add_head_not_allowed_methods_and_options_methods
           routes = endpoint.routes
           routes.each do |route|
             # using the :any shorthand produces [nil] for route methods, substitute all manually
-            route_key = route.pattern.to_regexp
-            routes_map[route_key] ||= {}
-            route_settings = routes_map[route_key]
-            route_settings[:pattern] = route.pattern
-            route_settings[:requirements] = route.requirements
-            route_settings[:path] = route.origin
+            route_settings = routes_map[route.pattern.to_regexp] ||= {}
+
+            route_settings.merge!(options: {},
```

Comment:
This is a bit hard to grok IMO. Feels like we should decide whether we want to use `merge!` or assign fields. Are you trying to optimize because `merge` can do things in bulk? I would `merge` everything or assign fields one by one. I also really dislike the side effect of `||=` and then taking a `route_settings` reference on that.

Reply:
Totally agree. I've initially re-wrote it to try to understand what it does also RuboCop had some issues with the length of the method when I added that one `:options` key into the hash.

So now as a second attempt to understand what it does i re-wrote it again :).